src/main/java/lotto/domain/number/LottoNumbers.java

```diff
@@ -0,0 +1,66 @@
+package lotto.domain.number;
+
+import lotto.domain.ticket.WinningLottoTicket;
+import lotto.exception.ErrorMessage;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class LottoNumbers {
+
+    private static final String LOTTO_NUMBER_DELIMITER = ",";
+    public static final int LOTTO_NUMBER_SIZE = 6;
+
+    private Set<LottoNumber> numbers;
+
+    private LottoNumbers() {
+    }
+
+    private LottoNumbers(final List<LottoNumber> numbers) {
+        validateNumbers(numbers);
+        this.numbers = new LinkedHashSet<>(numbers);
+    }
+
+    public static LottoNumbers auto() {
+        return new LottoNumbers(LottoNumbersGenerator.generate());
+    }
+
+    public static LottoNumbers of(final String numbers) {
+        return new LottoNumbers(Arrays.stream(numbers.split(LOTTO_NUMBER_DELIMITER))
+                .map(String::trim)
+                .map(Integer::parseInt)
+                .map(LottoNumber::of)
+                .collect(Collectors.toList()));
+    }
+
+    private void validateNumbers(final List<LottoNumber> numbers) {
+        if ((numbers == null) || (numbers.size() != LOTTO_NUMBER_SIZE)) {
+            throw new IllegalArgumentException(String.format(ErrorMessage.REQUIRED_SIZE_OF_LOTTO_NUMBERS, LOTTO_NUMBER_SIZE));
+        }
+        checkDuplication(numbers);
+    }
+
+    private void checkDuplication(final List<LottoNumber> numbers) {
+        Set<LottoNumber> numbersOfNotDuplicated = new HashSet<>(numbers);
+
+        if (numbersOfNotDuplicated.size() < LOTTO_NUMBER_SIZE) {
+            throw new IllegalArgumentException(ErrorMessage.DUPLICATED_NUMBERS);
+        }
+    }
+
+    public int matchCount(final WinningLottoTicket winningLottoTicket) {
+        return Math.toIntExact(numbers.stream()
+                .filter(winningLottoTicket::contains)
+                .count());
+    }
```

Comment:
```suggestion
    public int matchCount(final LottoNumbers winningLottoNumbers) {
        return Math.toIntExact(numbers.stream()
                .filter(winningLottoNumbers::contains)
                .count());
    }
```
요로면 LottoTicket이랑 WinningLottoTicket에 존재하는 contains 함수 제거 가능!

Reply:
상속을 이용!

`protected final LottoNumbers lottoNumbers;` 로 접근 지정자를 변경하면 
자식 클래스에서는 필드에 직접 접근이 가능
LottoNumbers의 contains 로 matchCount 를 구할 수 있다. (LottoTicket, WinningLottoTicket의 contains 제거 가능)

[관련 리뷰 1](https://github.com/zuniors/java-lotto/pull/2#discussion_r429363335)
[관련 리뷰 2](https://github.com/zuniors/java-lotto/pull/2#discussion_r429365279)

훨씬 더 깔끔해지네요~ 👍